src/main/java/seedu/address/model/consultation/Consultation.java

```diff
@@ -0,0 +1,104 @@
+package seedu.address.model.consultation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import seedu.address.model.student.Student;
+
+/**
+ * Represents a Consultation in the system.
+ * Guarantees: details are present and not null, field values are validated.
+ */
+public class Consultation {
+
+    // Identity fields
+    private final Date date;
+    private final Time time;
+
+    // Data field
+    private final List<Student> students;
```

Comment:
I'm unsure what "Identity" & "Data" Field refer to. Consider modifying the comments to make it more understandable.